### PR TITLE
Moving processes from MeshingApplication to cpp + cleaning + tweak

### PR DESCRIPTION
**Pull request by loumalouomega**
Additionally I added a check in the tests after the removing of the TetGen libray

### CODE REVIEW

File: applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.cpp [hunk 1]
```diff
@@ -0,0 +1,553 @@
...
+    GeometryData::IntegrationMethod this_integration_method;
+
+    // Iterate in the elements
+    ElementsArrayType& p_elements = ThisModelPart.Elements();
+    int num_elements = ThisModelPart.NumberOfElements();
+
+    const ProcessInfo& current_process_info = ThisModelPart.GetProcessInfo();
+
+    // Creating a buffer for parallel vector fill
+    const unsigned int num_threads = omp_get_max_threads();
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.cpp]: use OMPUtils

File: applications/MeshingApplication/custom_processes/metrics_hessian_process.h [hunk 2]
```diff
@@ -328,211 +259,23 @@ class ComputeHessianSolMetricProcess
...
-                const bounded_matrix<double, 3, 3> Hessian = prod(trans(DN_DX), values); 
-                const Vector HessianCond = MetricsMathUtils<3>::TensorToVector(Hessian);
-                
-                for(unsigned int iNode = 0; iNode < geom.size(); iNode++)
-                {
-                    for(unsigned int k = 0; k < 6; k++)
-                    {
-                        double& val = geom[iNode].GetValue(AUXILIAR_HESSIAN)[k];
-                        
-                        #pragma omp atomic
```
Comment (RiccardoRossi) [inline applications/MeshingApplication/custom_processes/metrics_hessian_process.h:492]: i see this is being removed, however you should have used a reduction for this

**Review by RiccardoRossi — APPROVED** [review]
I am approving, however please take a look at my comments prior to merging

**Comment by roigcarlo** [comment]
I would move the Constructor / Destructors to cpp as well. For example: https://github.com/KratosMultiphysics/Kratos/blob/adb71301bc2a3dc5b42cbdbe5736c18cadfe671c/applications/MeshingApplication/custom_processes/internal_variables_interpolation_process.h#L107-L125

**Comment by loumalouomega** [comment]
Okay

**Comment by pooyan-dadvand** [comment]
:+1: for moving implementations to cpp

**Pull request merged**